python/triton_kernels/triton_kernels/matmul_details/opt_flags_details/opt_flags_nvidia.py

```diff
@@ -36,14 +37,19 @@ def compute_block_k(m: int, k: int | None, is_persistent: bool, lhs_dtype, rhs_d
     # block_k needs to match the cacheline size (1024 bits)
     block_k = int(1024 // min(lhs_width, rhs_width))
     has_native_mxfp = target_info.cuda_capability_geq(10, 0)
+    has_mx_act_scale = precision_config is not None and precision_config.a_mx_scale is not None
     if rhs_width == 4 and not has_native_mxfp:
         block_k = 128
-    elif k is not None:
+    elif is_persistent and has_mx_act_scale and isinstance(precision_config.a_mx_scale.storage.layout,
```

Comment:
Unfortunately, we still need to handle cases when ``a_mx_scale`` is a plain torch Tensor. We should canonicalize this to simplify the code.
```suggestion
    has_swizzled_mx_act_scale = precision_config is not None and isinstance(precision_config.a_mx_scale, triton_kernels.tensor.Tensor)
if rhs_width == 4 and not has_native_mxfp:
block_k = 128
    elif is_persistent and has_swizzled_mx_act_scale and isinstance(precision_config.a_mx_scale.storage.layout,
```